### PR TITLE
Use engagementRate and display it as a percentage.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

As discussed on Slack we need to change from using the `engagedSessions` to the `engagementRate` metric, and display it as a percentage.

Addresses issue:

- #6219 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/utils/data-mock.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ const ANALYTICS_4_METRIC_TYPES = {
 	conversions: 'TYPE_INTEGER',
 	screenPageViews: 'TYPE_INTEGER',
 	engagedSessions: 'TYPE_INTEGER',
+	engagementRate: 'TYPE_INTEGER',
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics-4/utils/data-mock.js:46]: See https://github.com/google/site-kit-wp/pull/6689#discussion_r1127029972

File: assets/js/modules/analytics/components/module/ModulePopularPagesWidgetGA4.js [hunk 2]
```diff
@@ -188,7 +188,7 @@ function ModulePopularPagesWidgetGA4( props ) {
 			hideOnMobile: true,
 			field: 'metricValues.2.value',
 			Component: ( { fieldValue } ) => (
-				<span>{ numFmt( fieldValue, { style: 'decimal' } ) }</span>
+				<span>{ numFmt( Number( fieldValue ) / 100, '%' ) }</span>
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/components/module/ModulePopularPagesWidgetGA4.js:191]: Should we maybe use `parseFloat` instead?

Also, `fieldValue` for `engagementRate` will be a number between 0 and 1 (inclusive) so we shouldn't need to divide by 100.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @techanvil – just a few small points to address 👍

File: assets/js/modules/analytics-4/utils/data-mock.js (hunk 1 above)
Comment (techanvil) [reply assets/js/modules/analytics-4/utils/data-mock.js]: Thanks, I've fix this on the #6689 branch and cherry-picked the commit here as well.

File: assets/js/modules/analytics/components/module/ModulePopularPagesWidgetGA4.js (hunk 2 above)
Comment (techanvil) [reply assets/js/modules/analytics/components/module/ModulePopularPagesWidgetGA4.js]: It doesn't look like we need `parseFloat()` here, just passing `fieldValue` in to `numFmt()` without dividing it, while specifying the `%` format seems fine. Note that we pass `fieldValue` directly to `numFmt()` in the other cases too, and these are all copied from the UA version of the widget.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks!

**Pull request merged**